File: TypeScript/1. Two Sum.ts
// 1. Two Sum
// Level: Easy
// https://leetcode.com/problems/two-sum/

function twoSum(nums: number[], target: number): number[] {
  let map : {[key: number]: number} = {}
  for(let i = 0; i < nums.length; i++)
  {
    debugger;
    if(nums[i] in map){
      return [i, map[nums[i]]];
    }
    map[target - nums[i]] = i;
  }
  return [-1, -1];
};

File: Python/1. Two Sum.py
# 1. Two Sum
# Level: Easy
# https://leetcode.com/problems/two-sum/

class Solution:
  def twoSum(self, nums: List[int], target: int) -> List[int]:
    map = dict()
    for idx, x in enumerate(nums):
      if x in map:
        return [idx, map[x]]
      map[target - x] = idx

    return [-1, -1]